src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -698,6 +704,13 @@ private void AddTypeKind(INamedTypeSymbol symbol)
                     }
                 }
             }
+
+            static bool hasCloneMethod(INamedTypeSymbol symbol)
+            {
+                HashSet<DiagnosticInfo> ignored = null;
+                return symbol is Symbols.PublicModel.NamedTypeSymbol namedTypeSymbol &&
```

Comment:
>symbol is Symbols.PublicModel.NamedTypeSymbol namedTypeSymbol [](start = 23, length = 61)

This looks wrong. Please add a test that imports a record to VB, gets a symbol for it (VB symbol) and pushes it through here by feeding  the symbol to C# symbol display. This is a supported scenario.